external_libraries/amgcl/mpi/distributed_matrix.hpp

```diff
@@ -1240,7 +1240,7 @@ spectral_radius(const mpi::distributed_matrix<Backend> &A, int power_iters = 0)
 
                 for(size_t i = 0, m = C.send.count(); i < m; ++i)
                     b0_send[i] = b0[C.send.col[i]];
-                C.exchange(&b0_send[0], &b0_recv[0]);
+                C.exchange(b0_send.data(), b0_recv.data());
```

Comment:
maybe better to avoid cosmetic changes in this PR
is this from upstream ?

Reply:
It is from upstream, but it still seems to be out of sync (the line number is different):
https://github.com/ddemidov/amgcl/blob/3a207307f9b567079db6e3476c4258aad106a071/amgcl/mpi/distributed_matrix.hpp#L1224

Reply:
@RiccardoRossi can you then first make a PR with the upstream changes?